tests/chainer_tests/links_tests/connection_tests/test_convolution_2d.py

```diff
@@ -5,137 +5,101 @@
+    def generate_params(self):
+        initialW = chainer.initializers.Normal(1, self.W_dtype)
+        initial_bias = chainer.initializers.Normal(1, self.x_dtype)
+        return initialW, initial_bias
 
-    @attr.gpu
-    def test_col2im_consistency(self):
-        col = conv.im2col_cpu(self.x, 3, 3, 2, 2, 1, 1)
-        h, w = self.x.shape[2:]
-        im_cpu = conv.col2im_cpu(col, 2, 2, 1, 1, h, w)
-        im_gpu = conv.col2im_gpu(cuda.to_gpu(col), 2, 2, 1, 1, h, w)
-        testing.assert_allclose(im_cpu, im_gpu.get())
-
-    def check_forward_consistency(self):
-        x_cpu = chainer.Variable(self.x)
-        y_cpu = self.link(x_cpu)
-        self.assertEqual(y_cpu.data.dtype, self.x_dtype)
-
-        self.link.to_gpu()
-        x_gpu = chainer.Variable(cuda.to_gpu(self.x))
-        y_gpu = self.link(x_gpu)
-        self.assertEqual(y_gpu.data.dtype, self.x_dtype)
-
-        testing.assert_allclose(y_cpu.data, y_gpu.data.get())
-
-    @attr.gpu
-    @condition.retry(3)
-    def test_forward_consistency(self):
-        self.check_forward_consistency()
-
-    @attr.gpu
-    @condition.retry(3)
-    def test_forward_consistency_im2col(self):
-        with chainer.using_config('use_cudnn', 'never'):
-            self.check_forward_consistency()
-
-    def check_backward(self, x_data, y_grad):
-        gradient_check.check_backward(
-            self.link, x_data, y_grad, (self.link.W, self.link.b), eps=2 ** -3,
-            **self.check_backward_options)
-
-    @condition.retry(3)
-    def test_backward_cpu(self):
-        self.check_backward(self.x, self.gy)
-
-    @attr.gpu
-    @condition.retry(3)
-    def test_backward_gpu(self):
-        self.link.to_gpu()
-        self.check_backward(cuda.to_gpu(self.x), cuda.to_gpu(self.gy))
-
-    @attr.gpu
-    @condition.retry(3)
-    def test_backward_gpu_im2col(self):
-        self.link.to_gpu()
-        with chainer.using_config('use_cudnn', 'never'):
-            self.check_backward(cuda.to_gpu(self.x), cuda.to_gpu(self.gy))
+    def create_link(self, initializers):
+        initialW, initial_bias = initializers
 
-    def check_pickling(self, x_data):
-        x = chainer.Variable(x_data)
-        y = self.link(x)
-        y_data1 = y.data
-
-        del x, y
-
-        pickled = pickle.dumps(self.link, -1)
-        del self.link
-        self.link = pickle.loads(pickled)
-
-        x = chainer.Variable(x_data)
-        y = self.link(x)
-        y_data2 = y.data
+        link = links.Convolution2D(
+            3, 2, 3, stride=2, pad=1,
+            initialW=initialW,
+            initial_bias=initial_bias)
+
+        return link
+
+    def generate_inputs(self):
+        x = numpy.random.uniform(-1, 1,
+                                 (2, 3, 4, 3)).astype(self.x_dtype)
+        return x,
+
+    def forward_expected(self, link, inputs):
+        x, = inputs
+        y = link(x).array
+        return y,
+
+    def test_pickling(self, backend_config):
+        with backend_config:
```

Comment:
Shouldn't this test succeed without `with backend_config`?

Reply:
Yes, I added it in the first version of the test, it is corrected now.
Thanks!